Rename Processing Rules to Routing Rules (#28665)
DataDog/documentation: content/en/service_management/on-call/routing_rules.md

further_reading:
- link: '/service_management/on-call/'
  tag: 'Documentation'
  text: 'Datadog On-Call'
aliases:
    - /service_management/on-call/processing_rules/
---

{{< site-region region="gov" >}}
<div class="alert alert-warning">On-Call is not supported for your selected <a href="/getting_started/site">Datadog site</a> ({{< region-param key="dd_site_name" >}}).</div>
{{< /site-region >}}

## Overview

With routing rules, you can define granular logic to control how alerts reach your team. Instead of sending alerts through a single escalation policy, you can create flexible, condition-based rules to route them based on priority, time of day, tags, and more.

## Routing rules examples

- Route alerts by priority:
  - Send **priority 1** alerts to your primary escalation policy.
  - Send **priority 2–4** alerts to Slack or Microsoft Teams.

- Route alerts by time of day:
  - During business hours, route alerts to an escalation policy.
  - After hours, route critical alerts to paging, and non-critical alerts to chat.

- Use Dynamic Urgency to automatically detect urgency from the monitor alert:
  - `warn` status ➝ low urgency
  - `alert` status ➝ high urgency

>   The urgency of a page determines how end users are notified, based on their preferences. <br>

Reviewer: ```suggestion
  The urgency of a page determines how end users are notified, based on their preferences. <br> <br>
```

>   The urgency of a page determines how end users are notified, based on their preferences. <br> <br>

Reviewer: ```suggestion
  The urgency of a page determines how end users are notified, based on their preferences. <br><br>
```

**⚠️ Note:** Routing rule syntax is **case sensitive**. For example, `tags.env:Prod` will not match `tags.env:prod`.

**Supported attributes:**

| Attribute      | Description                                                                 | Example                                 |
|----------------|-----------------------------------------------------------------------------|-----------------------------------------|
| `tags`         | Tags on the incoming alert                                                  | `tags.env:prod`                         |
| `groups`       | Monitor group names                                                         | `groups:"service:checkout-service"`     |
| `priority`     | Monitor priority (1–5)                                                      | `priority:(1 OR 2)`                     |
| `alert_status` | Monitor status (`error`, `warn`, `success`)                                 | `alert_status:(error OR warn)`          |

<placeholder_screenshot>
_A screenshot showing an example routing rule with syntax input, including `priority:(1)` and `alert_status:error` in the query builder UI_

---

### Best Practices

> - Think in terms of **urgency vs visibility**:
>   - Use paging + escalation for urgent, high-impact alerts.
>   - Use Slack, or Teams for lower-severity, non-paging notifications.

Reviewer: ```suggestion
- Balance visibility with urgency:
  - Use paging and escalation policies for critical alerts that require immediate action.
  - Use Slack or Teams for lower-severity issues that need awareness but don’t warrant an on-call response.
```
